src/elisp/treemacs-mode.el

```diff
@@ -328,9 +328,10 @@ If there is no node at point use \"~/\" instead.
 
 Used as a post command hook."
   (-if-let* ((btn  (treemacs-current-button))
+             (project (treemacs-project-of-node btn))
              (path (or (treemacs-button-get btn :default-directory)
                        (treemacs--nearest-path btn))))
-      (when (and (stringp path)
+      (when (and (treemacs-project->is-readable? project)
```

Comment:
Here's a bug. Checking that the project is readable is not sufficient, since projects may contain project/dir extensions and tag nodes.

Reply:
Fixed this one by making `treemacs--nearest-path` skip extension paths. I also changed the function into a recursive implementation, simplifying it. I believe it won't cause problems, since the number of skipped non-file nodes is likely small (until someone makes a hugely nested extension.)